In [11]:
import pandas as pd

df = pd.read_json(r"C:\Users\jorge leandro\OneDrive\Desktop\proyecto_final_md\data\raw\streaming_users_dirty.json")

In [12]:
#Visualización de los primeros registros
df.head()


,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1


In [13]:
#Dimensiones del dataset
df.shape


(8160, 8)

In [14]:
#Información general
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   object 
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   object 
 5   favorite_genre            7920 non-null   object 
 6   last_login_date           7840 non-null   object 
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), object(4)
memory usage: 510.1+ KB


In [15]:
#Tipos de datos
df.dtypes


user_id                       int64
age                           int64
subscription_plan            object
monthly_watch_time_mins     float64
country                      object
favorite_genre               object
last_login_date              object
customer_support_tickets      int64
dtype: object

In [16]:
#Estadísticas descriptivas
df.describe()


,user_id,age,monthly_watch_time_mins,customer_support_tickets
count,8160.000000,8160.000000,7967.000000,8160.000000
mean,13995.433824,34.096814,1107.346153,1.800980
std,2310.810660,14.511304,5310.442604,11.334969
min,10000.000000,-5.000000,-120.000000,-1.000000
25%,11987.750000,25.000000,489.200000,0.000000
50%,13998.500000,33.000000,757.400000,1.000000
75%,15997.250000,42.000000,1045.700000,1.000000
max,17999.000000,150.000000,99999.000000,150.000000


In [17]:
#Valores faltantes
df.isnull().sum()


user_id                       0
age                           0
subscription_plan             0
monthly_watch_time_mins     193
country                       0
favorite_genre              240
last_login_date             320
customer_support_tickets      0
dtype: int64

In [18]:
#Porcentaje de valores nulos en cada columna
(df.isnull().sum() / len(df)) * 100


user_id                     0.000000
age                         0.000000
subscription_plan           0.000000
monthly_watch_time_mins     2.365196
country                     0.000000
favorite_genre              2.941176
last_login_date             3.921569
customer_support_tickets    0.000000
dtype: float64

In [19]:
#Registros duplicados
df.duplicated().sum()


np.int64(126)

Este dataset representa:

Información de usuarios de una plataforma de streaming: edad, país, plan, hábitos de consumo, género favorito, actividad reciente y reclamos al soporte.
Cuenta con 8160 registros. 
Existen valores faltantes en las columnas last_login_date, montly_watch_time_mins y favorite_genre; y con 126 registros duplicados 